.github/actions/gradle/experiment-1/action.yml

```diff
@@ -42,31 +42,36 @@ runs:
       run: |
         cd gradle-enterprise-gradle-build-validation
 
-        EXTRA_ARGS=""
+        # Need to process separately tasks argument as it may contain whitespaces
+        ARG_TASKS=""
+        if [ ! -z "${{ inputs.tasks }}" ]; then
+          ARG_TASKS="-t ${{inputs.tasks }}"
+        fi
+
+        # Unprotected arguments
+        UNPROTECTED_ARGS="-r ${{ inputs.gitRepo }}"
         if [ ! -z "${{ inputs.gitBranch }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -b ${{ inputs.gitBranch }}"
+          UNPROTECTED_ARGS="$UNPROTECTED_ARGS -b ${{ inputs.gitBranch }}"
         fi
         if [ ! -z "${{ inputs.gitCommitId }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -c ${{ inputs.gitCommitId }}"
-        fi
-        if [ ! -z "${{ inputs.projectDir }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -p ${{ inputs.projectDir }}"
-        fi
-        if [ ! -z "${{ inputs.tasks }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -t ${{ inputs.tasks }}"
+          UNPROTECTED_ARGS="$UNPROTECTED_ARGS -c ${{ inputs.gitCommitId }}"
         fi
+        ARG_ARGS="${{ inputs.args }}"
         if [ ! -z "${{ inputs.args }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -a ${{ inputs.args }}"
+          UNPROTECTED_ARGS="$UNPROTECTED_ARGS -a ${{ inputs.args }}"
+        fi
+        if [ ! -z "${{ inputs.projectDir }}" ]; then
+          UNPROTECTED_ARGS="$UNPROTECTED_ARGS -p ${{ inputs.projectDir }}"
         fi
         if [ ! -z "${{ inputs.gradleEnterpriseUrl }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -s ${{ inputs.gradleEnterpriseUrl }}"
+          UNPROTECTED_ARGS="$UNPROTECTED_ARGS -s ${{ inputs.gradleEnterpriseUrl }}"
         fi
         if [ ! -z "${{ inputs.enableGradleEnterprise }}" ]; then
-          EXTRA_ARGS="$EXTRA_ARGS -e"
+          UNPROTECTED_ARGS="$UNPROTECTED_ARGS -e"
         fi
 
         # run experiment
-        ./01-validate-incremental-building.sh -r ${{ inputs.gitRepo }} $EXTRA_ARGS
+        ./01-validate-incremental-building.sh "${ARG_TASKS}" ${UNPROTECTED_ARGS}
```

Comment:
Directories can also contain whitespace.

Reply:
Isn't there a way to "protect" the arguments of each option separately? It does not seem "right" that we do this special casing just for tasks, and that it percolates all the way to cmd invocation here.

Reply:
I see. It might be better to add each option separately, always protected by quotes, just like you did for tasks, but also for all other options.

Reply:
> It might be better to add each option separately, always protected by quotes

Yeah, I agree. That would also make it much easier to see how each argument is quoted, which is difficult to see and understand in the current implementation.